controllers/managedocs_controller_test.go

```diff
@@ -839,6 +860,60 @@ var _ = Describe("ManagedOCS controller", func() {
 				utils.WaitForResource(k8sClient, ctx, amConfigSecretTemplate.DeepCopy(), timeout, interval)
 			})
 		})
+		When("there is grafana-datasources secret in openshift-monitoring namespace", func() {
+			It("should create k8s-metrics-service-monitor-auth secret in primary namespace", func() {
+				// Create grafana-datasources secret
+				authInfoStructure := struct {
+					DataSources []struct {
+						BasicAuthPassword string `json:"basicAuthPassword"`
+						BasicAuthUser     string `json:"basicAuthUser"`
+					} `json:"datasources"`
+				}{}
+				authInfoStructure.DataSources = make([]struct {
+					BasicAuthPassword string "json:\"basicAuthPassword\""
+					BasicAuthUser     string "json:\"basicAuthUser\""
+				}, 1)
+				authInfoStructure.DataSources[0].BasicAuthPassword = "password"
+				authInfoStructure.DataSources[0].BasicAuthUser = "internal"
+				grafanaSecret := grafanaDatasourceSecretTemplate.DeepCopy()
+				grafanaSecret.Data = make(map[string][]byte)
+				grafanaSecret.Data["prometheus.yaml"], _ = json.Marshal(authInfoStructure)
```

Comment:
All of this code should go into the `grafanaDatasourceSecretTemplate` template, the idea behind templates is that you should not need to refill static info every time manually. 

```suggestion
				grafanaSecret := grafanaDatasourceSecretTemplate.DeepCopy()
```